x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/enrollment/tool/BootstrapPasswordAndEnrollmentForInitialNode.java

```diff
@@ -0,0 +1,196 @@
+    }
+    KeyStoreWrapper getKeyStoreWrapper(Environment env) throws Exception { return keyStoreFunction.apply(env); }
+    OptionParser getParser() { return parser; }
+
+    BootstrapPasswordAndEnrollmentForInitialNode() {
+        this(
+            environment -> new CommandLineHttpClient(environment),
+            environment -> KeyStoreWrapper.load(environment.configFile()),
+            environment -> new CreateEnrollmentToken(environment)
+        );
+    }
+
+    BootstrapPasswordAndEnrollmentForInitialNode(Function<Environment, CommandLineHttpClient> clientFunction,
+                                                 CheckedFunction<Environment, KeyStoreWrapper, Exception> keyStoreFunction,
+                                                 CheckedFunction<Environment, CreateEnrollmentToken, Exception>
+                                                     createEnrollmentTokenFunction){
+        super(clientFunction, keyStoreFunction, "Set elastic password and generate enrollment token for initial node");
+        this.createEnrollmentTokenFunction = createEnrollmentTokenFunction;
+        parser.allowsUnrecognizedOptions();
+    }
+
+    public static void main(String[] args) throws Exception {
+        exit(new BootstrapPasswordAndEnrollmentForInitialNode().main(args, Terminal.DEFAULT));
+    }
+
+    @Override
+    protected void execute(Terminal terminal, OptionSet options, Environment env) throws Exception {
+        if (options.nonOptionArguments().contains("--explicitly-acknowledge-execution") == false) {
+            throw new UserException(ExitCodes.CONFIG, "This command is not intended for end users");
+        }
+        if (env.settings().hasValue(XPackSettings.ENROLLMENT_ENABLED.getKey()) && false ==
+            XPackSettings.ENROLLMENT_ENABLED.get(env.settings())) {
+            throw new UserException(ExitCodes.NOOP, "Enrollment is explicitly disabled.");
+        }
+        final CommandLineHttpClient client = getClient(env);
+        final CreateEnrollmentToken createEnrollmentToken = getCreateEnrollmentToken(env);
+        keyStoreWrapper = getKeyStoreWrapper(env);
+        ReadBootstrapPassword(env, terminal);
+        checkClusterHealthWithRetries(client, terminal, elasticUser, credentialsPassword, 0, 5, false);
+        if (Strings.isNullOrEmpty(bootstrapPassword.toString())) {
+            changeElasticUserPassword(terminal, client);
+        } else {
+            password = bootstrapPassword;
+        }
+        token = createEnrollmentToken.createKibanaEnrollmentToken(elasticUser, password);
+        fingerprint = createEnrollmentToken.getFingerprint();
+
+        if (Strings.isNullOrEmpty(bootstrapPassword.toString())) {
+            terminal.println("'elastic' user password: " + password);
+        }
+        terminal.println("CA fingerprint: " + fingerprint);
+        terminal.println("Kibana enrollment token: " + token);
+        if (options.nonOptionArguments().contains("--docker")) {
```

Comment:
This doesn't need to be a nonOption. We can have a proper option and it will be set by the startup script ( assuming that you have discussed this before and we have a way to determine that we are running in docker that we plan on using )

Reply:
This is not addressed @BigPandaToo